### PR TITLE
feat(aws): adding sns/sqs utils

### PR DESCRIPTION
**Pull request by emjburns**
Pulling sns and sqs management to Kork for ease of use across services.
Fixing permissions on sqs creation to be more narrow.

### CODE REVIEW

File: kork-aws/src/main/groovy/com/netflix/spinnaker/kork/aws/pubsub/PubSubUtils.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+   * This policy allows messages to be sent from an SNS topic.
+   */
+  static Policy buildSQSPolicy(ARN queue, ARN topic) {
+    Statement snsStatement = new Statement(Statement.Effect.Allow).withActions(SQSActions.SendMessage);
+    snsStatement.setPrincipals(Principal.All);
+    snsStatement.setResources(Collections.singletonList(new Resource(queue.getArn())));
+    snsStatement.setConditions(Collections.singletonList(
+      new Condition().withType("ArnEquals").withConditionKey("aws:SourceArn").withValues(topic.getArn())
+    ));
+
```
Comment (emjburns) [inline kork-aws/src/main/groovy/com/netflix/spinnaker/kork/aws/pubsub/PubSubUtils.java]: Removed the wide open policy of anyone can send to this queue (was previously right here).

File: kork-aws/src/main/groovy/com/netflix/spinnaker/kork/aws/ARN.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+public class ARN {
+
+  static final Pattern PATTERN = Pattern.compile("arn:aws(?:-cn|-us-gov)?:.*:(.*):(\\d+):(.*)");
+
+  private String arn;
+  private String region;
+  private String name;
+  private String account;
+
+  ARN(String arn) {
```
Comment (robzienert) [inline kork-aws/src/main/groovy/com/netflix/spinnaker/kork/aws/ARN.java]: Should be `public`.

File: kork-aws/src/main/groovy/com/netflix/spinnaker/kork/aws/pubsub/PubSubUtils.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+import java.util.Collections;
+import java.util.HashMap;
+
+/**
+ * Utils for working with AWS SNS and SQS across services
+ */
+public class PubSubUtils {
+  private static final Logger log = LoggerFactory.getLogger(PubSubUtils.class);
+
+  static String ensureQueueExists(AmazonSQS amazonSQS,
```
Comment (robzienert) [inline kork-aws/src/main/groovy/com/netflix/spinnaker/kork/aws/pubsub/PubSubUtils.java]: `public`

Comment (robzienert) [reply kork-aws/src/main/groovy/com/netflix/spinnaker/kork/aws/pubsub/PubSubUtils.java]: And also alignment on method args. (for this method and others)

**Pull request merged**